### PR TITLE
What's an HTTP/3 connection?

### LINKED ISSUE #4206: HTTP-31: HTTP/3 Connection definition?
Section 2.2.:

"connection:
    A transport-layer connection between two endpoints, using QUIC as the transport protocol.
connection error:
    An error that affects the entire HTTP/3 connection."

In a total of 11 places this document uses "HTTP/3 connection". The word connection exists 213 times in this document (HTML version). So are there a difference between connection and HTTP/3 Connection, and if it is can that be clearly expressed. Else maybe alignment would be better.

Secondly, I think the above "connection" definition is likely missing one important aspect isn't it? I have the impression that an "HTTP/3" connection is a QUIC connection established with the purpose of communicating HTTP/3 over it. That intention is likely explicitly expressed at QUIC connection establishment using ALPN?

### PR DESCRIPTION
**Pull request by MikeBishop**
Fixes #4206.

### CODE REVIEW

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -241,6 +241,9 @@ frame:
   frames."  References without this preface refer to frames defined in
   {{frames}}.
 
+HTTP/3 connection:
+: A QUIC connection where the negotiated application protocol is HTTP/3.
```
Comment (LPardue) [inline draft-ietf-quic-http.md:245]:
```suggestion
: A QUIC connection where the selected application protocol is HTTP/3.
```

make it more consistent with the notion of H3 selection in #4211

Comment (ianswett) [reply draft-ietf-quic-http.md:245]: Huh, I always thought negotiated was preferred since we're using ALPN?

**Review by ianswett — APPROVED** [review]

File: draft-ietf-quic-http.md (hunk 1 above)
Comment (LPardue) [reply draft-ietf-quic-http.md:245]: The spec also says that other methods of selecting H3 are possible. And if we're speaking generically, then that might not be a negotiation. But I'll let Mike make the editorial decision here.

Comment (martinthomson) [reply draft-ietf-quic-http.md:245]: Either works; but as ALPN involves a server selection and the N stands for negotiation, I don't think it matters.

**Review by martinthomson — APPROVED** [review]

**Review by gloinul — COMMENTED** [review]
As I said in the Issue there likely need to be a review of the 200+ times the word "connection" is used in the document to determine if one need to stress that it is the HTTP/3 connection or some other connection.

File: draft-ietf-quic-http.md (hunk 1 above)
Comment (MikeBishop) [reply draft-ietf-quic-http.md:245]: I'm inclined toward keeping negotiation here.

File: draft-ietf-quic-http.md [hunk 2]
```diff
@@ -1052,7 +1057,7 @@ the server.  For more detail on QUIC streams, see Section 2 of
 When HTTP fields and data are sent over QUIC, the QUIC layer handles most of
 the stream management.  HTTP does not need to do any separate multiplexing when
 using QUIC - data sent over a QUIC stream always maps to a particular HTTP
-transaction or connection context.
+transaction or to the HTTP/3 connection context as a whole.
```
Comment (janaiyengar) [inline draft-ietf-quic-http.md:1060]: suggestion:
```suggestion
transaction or to the entire HTTP/3 connection context.
```

File: draft-ietf-quic-http.md [hunk 3]
```diff
@@ -1561,8 +1567,8 @@ type H3_ID_ERROR.
 In the client to server direction, the GOAWAY frame carries a Push ID encoded as
 a variable-length integer.
 
-The GOAWAY frame applies to the connection, not a specific stream.  A client
-MUST treat a GOAWAY frame on a stream other than the control stream as a
+The GOAWAY frame applies to the connection as a whole, not a specific stream.  A
```
Comment (janaiyengar) [inline draft-ietf-quic-http.md:1570]:
```suggestion
The GOAWAY frame applies to the entire connection, not a specific stream.  A
```

**Review by janaiyengar — APPROVED** [review]
a couple of suggestions

**Pull request merged**